### PR TITLE
OpenCAD 1.0.0 Extravaganza

### LINKED ISSUE #366: Table Columns on Civlian Dashboard
**Describe the bug**
When logged in as a civilian and you've created and the "My Identities" table has one too many columns in the content than the header.
This is caused by the function ncicGetNames() in civActions.php between lines 124 and 162.

**To Reproduce**
1. Go to Civilian
2. Create a new identity
3. View the new Identity under "My Identities"

**Expected behavior**
The table header has the same column count as the content.

**Screenshots**
No Screenshots since this should be easy to recreate

**Desktop (please complete the following information):**
 - Browser [Brave]

### LINKED ISSUE #367: Driver License assign when Identity creation
**Describe the bug**
It is not possible to select valid from the list, it will automatically select the default MySQL "Unobtained" no matter what.
Also the PHP function used to split the dataset from the database on line 57 in publicFunctions.php is returning an array where the first (0) is and I quote ",'valid"
May have something to do with the trim.

**To Reproduce**
1. Go to Civilian Dashboard
2. Click on Create an Identity
3. Scroll down to Driver License Status
4. Select anything but Unobtained.

**Expected behavior**
That a user can set the specific status of their driver license

**Desktop (please complete the following information):**
 - Browser [Brave]

**Additional context**
It is not possible to change status either after this.
It posts the error that the user already exists.

### LINKED ISSUE #374: Proposal to drop support for old versions of PHP.
**Synopsis**
OpenCAD's officially supports PHP 7.3 (currently recommended release) and PHP 7.2 (minimum) per our current documentation.
Running older versions of PHP may expose servers to security vulnerabilities and children eating bugs that would have otherwise already been fixed on current PHP branches. 

PHP 7.2 (OpenCAD's minimum supported PHP branch) has reached [end of life](https://www.php.net/eol.php) and received its last update on 11/30/2020

PHP 7.3 while not end of life is [no longer actively supported](https://www.php.net/supported-versions.php) and will only receive critical security fixes going forward (EOL on 12/6/2021).

**Where do we begin?**
We need to first verify OpenCAD's compatibility with the currently "maintained" branches of PHP (E.g., "7.3, 7.4, and 8.0") before making any further decisions. Should everything work right of the get go, we can proceed directly to receiving community feedback. Should things not be working as expected, we should go ahead and opened the relevant bug reports and pull requests without introducing regressions if possible.

**What do we do after?**
Once we shift OpenCAD's officialy supported branches of PHP to maintained versions we should begin to follow PHP's [active support window](https://www.php.net/supported-versions.php) going forward.

**What other considerations must we discuss before making any changes?**
While this is a minor change in principle, the fact is that there are so many different environments being utilized by our community. 

Unfortunately, a good amount of shared hosting providers does not offer the currently supported versions of PHP to their customers for a variety of reasons.

***

The PHP Group. (n.d.). PHP: Supported Versions. PHP. Retrieved March 28, 2021, from https://www.php.net/supported-versions.php

The PHP Group. (n.d.-b). PHP: Unsupported Branches. PHP. Retrieved March 28, 2021, from https://www.php.net/eol.php

### LINKED ISSUE #378: Spelling: Verstion -> Version
Spelling mistake.
```suggestion
* @since OpenCAD version 1.0
```

_Originally posted by @ItsAGeekThing in https://github.com/opencad-app/OpenCAD-php/pull/376#discussion_r606023669_

### LINKED ISSUE #380: White container showing on oc-start when LIVEMAP_URL isn't defined
**Describe the bug**
oc-start shows a white "tail" on the LiveMap/Patrol Information section of the dashboard.

*This does not effect a single browser nor is it influenced by a given browser. This is a UI tweak to clean it up.*

**To Reproduce**
1. Install source from #376.
2. Login

**Expected behavior**

**Screenshots**
* Screenshot

### LINKED ISSUE #383: Make sure that this logger's configuration is safe.
**Describe the bug**
Make sure that this logger's configuration is safe.

**To Reproduce**
No EU impact. The code is internal to OC.

**Expected behavior**
Loggers shouldn't be configured out side of OC_DEBUG being true.

**Additional context**
[SonarCloud issue](https://sonarcloud.io/project/security_hotspots?id=opencad-app_OpenCAD-php&hotspots=AXh7qemJiIV3naYkCyql)
[SonarCloud issue](https://sonarcloud.io/project/security_hotspots?id=opencad-app_OpenCAD-php&hotspots=AXh7qekZiIV3naYkCym0)
[SonarCloud issue](https://sonarcloud.io/project/security_hotspots?id=opencad-app_OpenCAD-php&hotspots=AXh7qepyiIV3naYkCyyH)

### LINKED ISSUE #384: Proposal to drop support for old versions of SQL.
**Synopsis**
OpenCAD's officially supports MySQL >=5.5 (no recommended version) and MariaDB >=10.0 (with 10.3 being recommended) per our current documentation.
Running older versions of MySQL/MariaDB may expose servers to security vulnerabilities and children eating bugs that would have otherwise already been fixed on current releases. 

MySQL 5.5 (OpenCAD's minimum supported version) has reached [end of life](https://www.oracle.com/us/support/library/lifetime-support-technology-069183.pdf) (support ended on Dec. 2015).
MySQL 5.6's support ended in Feb. 2018 and MySQL 5.7's in Oct. 2020.

_The latest version of MySQL that currently has support is **8.0** (which ends in April 2023)._

MariaDB 10.0 (OpenCAD's minimum supported version) has reached [end of life](https://mariadb.org/about/#maintenance-policy) (support ended on Mar. 2019).
MariaDB 10.1's support ended in Oct. 2020.

_The latest version of MariaDB that currently has support is **10.2** (which ends in May 2022)_

**Where do we begin?**
We should set our minimal (and supported) version of MySQL as 8.0 (support end in April 2023).
We should set our minimal version of MariaDB as 10.2 (support end in May 2022) and the recommended version as 10.5 (support ends in June 2025).

**What do we do after?**
Once we shift OpenCAD's officially supported version of MySQL and MariaDB to maintained versions we should begin to follow [Oracle's Oracle Lifetime Support Policy](https://www.oracle.com/us/support/library/lifetime-support-technology-069183.pdf) and [MariaDB's Maintenance policy](https://mariadb.org/about/#maintenance-policy), respectively.

**What other considerations must we discuss before making any changes?**
While this is a minor change in principle, the fact is that there are so many different environments being utilized by our community. 

***

_* We base MySQL's EOL date off of its support end date and not the extended support date (Oracle charges a [support fee for extended support](https://www.oracle.com/support/lifetime-support/) which is intended for their enterprise customers)._

### LINKED ISSUE #394: [PROPOSAL] Deprecate passwordReset column in users table in favour of adding a new status to approved.
**Describe the solution you'd like**
1. Deprecate the _passwordReset_ column in the users table. 
2. Rename the _approved_ column to _accountStatus_.
3. Introduce new status of 3 which would correspond to "User must reset password upon next login".

### PR DESCRIPTION
**Pull request by ItsAGeekThing**
**PR is a Work In Progress**
_Testing and documentation of bugs is both encouraged and appreciated._

This pull request implements the following changes:

* Removes commented out code.
    * Commented out code adds bloat and serves no use.
If it is needed in the future for whatever reason, it can be retrieved from Git history.
* Remove configuring loggers outside of Debug Mode.
    * There were a couple functions where it was possible for OpenCAD to configure loggers outside of Debug Mode.
* Resolve unsafe cross-origin.
    * Added `rel="noopener"` to the Stopwatch to prevents the external resource from abusing (not that they would) window.opener, potentially avoiding a redirect to a malicious URL if using a browser before Chrome 88, Firefox 79, or Safari 12.1.
* Implement SRI and remove legacy vendor example folders.
Subresource Integrity allows the browser to verify that resources hosted on an external server were not tampered with.
* Utilize HTTPS instead of HTTP in included libraries. 
* Starting to bring forward the theming the localization features that we already implemented.
    * Detaches theming from the core code.
    * Eliminates EOL vendor libraries.
* [#366] Combined dl_type and dl_status in to one column to equalize table content with header(s).
    * [#366] Moved Driver's License content to after Build as it doesn't describe an attribute of a person.
    * [#366] It was more consistent with the rest of the headers to have "Build" for the DL column.
    * Reported by @AniTexs.
* Removed SQL directory to reduce the number of places SQL templates need to be updated.
    * Updated oc-install to reference NEW location of sql scripts.
* [#367] Fixed trimming issue for dl_status selection.
    * Saving/Editing will be addressed in a future PR.
    * Also reported by @AniTexs.
 * Remove superfluous vendors folder. The libraries contaioned with in aren't use in core code but used in themes.
* Remove cad.php and civilian.php. …
* Database rewrite. Enough said.
* Introduce changes to oc-install.sql to be compliant with re-write.
* Add LIVE_MAP option to oc-config.sample.php.
* Update database schema. 
     * Required chamges for oc-update.sql forthcoming.
* Move functions in to oc-includes. 
    * Remove depecated actions folder.
* Update installer to support LIVEMAP_URL.
    * Update installer to support WEBHOOK_URL
* Remove superfluous $testing variable.
    * The subject of settings/defaults for testing needs to revisited.
* Add reference to oc_config in oc-functions to access CONSTANTS.
* Update internal libraries to use standard sql table naming conventions.
* Move debug info to a modal which is only accessible to admins.
    * This ensure debug info doesn't dsitrub EUs.
* Fix paths in mdt.php for NCIC tools. Strip "/js/" from path.
* Remove deprecated JIRA widget from ncicAdmin.php.
* Remove deprecated JIRA widget from error pages plugin. 
    * This code should be intefrated directly with the core codebase.
* Add dashboard.modal.inc.php for oc-start.php.
    * Contaions modal for debug output.
* Update .gitignore.
* [#379] Centralize and improve entropy implementation.
* [#380] Tweak code to only show content section when LIVEMAP_URL is defined in oc-config."
* [#374/#384] Drop support for end-of-life PHP and MySQL/MariaDB versions.
* [#389] Renamed weaponPermit to weaponPermitStatus
    * [#389] Tweaked weaponPermitType to align with new database spec.
    * [#389] Tweaked weaponPermitIssuedBy to align with the new database spec.
* Further modularize theming. Default now uses the "building blocks" in it's includes folder.
    * Themeing no longer relies on oc-includes.
* Add COMMUNITY_HOMEPAGE constant to oc-config.sample.php. 
    
### ***This PR requires testing. This can be done while in draft state.***

### CODE REVIEW

File: oc-config.sample.php [hunk 1]
```diff
@@ -759,6 +759,16 @@
 **/
 define('WEBHOOK_URL', '<WEBHOOK_URL>');
 
+/**#@+
+* LIVEMAP_URL
+* 
+* If configured a LiveMap will show on dashboard and throughout OpenCAD.
+* 
+* @since OpenCAD verstion 1.0
```
Comment (ItsAGeekThing) [inline oc-config.sample.php]: Spelling mistake.
```suggestion
* @since OpenCAD version 1.0
```

Comment (phillf) [reply oc-config.sample.php]: Addressed in #378.

File: oc-admin/dataManagement/radioCodeManager.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+				alert('Exeption:' + exception);
+			}
+		});
+	})
+	</script>
+</body>
+
+			<script type="text/javascript"
+		src="https://jira.opencad.io/s/a0c4d8ca8eced10a4b49aaf45ec76490-T/-f9bgig/77001/9e193173deda371ba40b4eda00f7488e/2.0.24/_/download/batch/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector.js?locale=en-US&collectorId=ede74ac1">
+	</script>
```
Comment (ItsAGeekThing) [inline oc-admin/dataManagement/radioCodeManager.php]:
```suggestion
```

Introduces obsolete Issue Collector from what was Jira.

File: oc-apps/civilian.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+		<?php
+        include ( ABSPATH . "/oc-includes/jquery-colsolidated.inc.php");
+        include ( ABSPATH . "/" . OCTHEMES . "/" . THEME . "/modals/civilian.modals.inc.php");
+        ?>
+        
+	</body>
+
+    <script type="text/javascript" src="<?php echo BASE_URL ."/". OCTHEMES ."/". THEME . "/js/civilian.inc.js";?>"></script>
+    <script type="text/javascript" src="https://jira.opencad.io/s/a0c4d8ca8eced10a4b49aaf45ec76490-T/-f9bgig/77001/9e193173deda371ba40b4eda00f7488e/2.0.24/_/download/batch/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector.js?locale=en-US&collectorId=ede74ac1"></script>
```
Comment (ItsAGeekThing) [inline oc-apps/civilian.php]:
```suggestion
```

Introduces obsolete Issue Collector from what was Jira.

File: oc-apps/oc-profile.php [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+		<!-- ./ modal-dialog modal-lg -->
+	</div>
+	<!-- ./ modal fade bs-example-modal-lg -->
+		<?php
+		include ( ABSPATH . "/oc-includes/jquery-colsolidated.inc.php"); ?>
+</body>
+
+						<script type="text/javascript"
+				src="https://jira.opencad.io/s/a0c4d8ca8eced10a4b49aaf45ec76490-T/-f9bgig/77001/9e193173deda371ba40b4eda00f7488e/2.0.24/_/download/batch/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector.js?locale=en-US&collectorId=ede74ac1">
+		</script>
```
Comment (ItsAGeekThing) [inline oc-apps/oc-profile.php]:
```suggestion
```

Introduces obsolete Issue Collector from what was Jira.

File: oc-apps/oc-supervisor/callhistory.php [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    $(document).ready(function() {
+
+      $('#pendingUsers').DataTable({
+        paging: false,
+        searching: false
+      });
+
+    });
+    </script>
+    <script type="text/javascript" src="https://jira.opencad.io/s/a0c4d8ca8eced10a4b49aaf45ec76490-T/-f9bgig/77001/9e193173deda371ba40b4eda00f7488e/2.0.24/_/download/batch/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector/com.atlassian.jira.collector.plugin.jira-issue-collector-plugin:issuecollector.js?locale=en-US&collectorId=ede74ac1"></script>
```
Comment (ItsAGeekThing) [inline oc-apps/oc-supervisor/callhistory.php]:
```suggestion
```
Introduces obsolete Issue Collector from what was Jira.

File: oc-content/plugins/error/index.php [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                                    Description: <?php echo $error; ?>
+                                </div>
+                                <?php if(!empty($error_blob)) { ?> 
+                                <div class="x_content">
+                                    PHP Exception: <?php echo $error_blob; ?>
+                                </div>
+                                <br><br>
+                                <?php } ?>
+                                <div class="x_content">
+                                    If this error persists, please raise up a <a rel="noopener" href="https://jira.opencad.io/secure/CreateIssue!default.jspa">ticket</a>.
```
Comment (ItsAGeekThing) [inline oc-content/plugins/error/index.php]:
```suggestion
                                    If this error persists, please raise up a <a rel="noopener" href="https://github.com/opencad-app/OpenCAD-php/issues/new/choose">bug report</a> on GitHub.
```

**Review by sudenoh — APPROVED** [review]

**Comment by sudenoh** [comment]
Hoping to see this come out without many issues.

**Comment by phillf** [comment]
@sudenoh Not quite there yet but testing and documenting of bugs would be greatly appreciated.

**Comment by phillf** [comment]
Per conversation with @ItsAGeekThing:

- Draft status removed.
- Review for canary should occur to simplify workflow and allow progress to take place in root repository.

***INSTABILITY EXPECTED***

**Review by phillf — APPROVED** [review]

**Pull request merged**